zerver/tests/test_events.py

```diff
@@ -2855,6 +2876,49 @@ def test_notify_realm_export(self) -> None:
         error = deletion_schema_checker('events[0]', events[0])
         self.assert_on_error(error)
 
+    def test_notify_realm_export_on_failure(self) -> None:
+        pending_schema_checker = self.check_events_dict([
+            ('type', equals('realm_export')),
+            ('exports', check_list(check_dict_only([
+                ('id', check_int),
+                ('export_time', check_float),
+                ('acting_user_id', check_int),
+                ('export_url', equals(None)),
+                ('deleted_timestamp', equals(None)),
+                ('failed_timestamp', equals(None)),
+                ('pending', check_bool),
+            ]))),
+        ])
+
+        failed_schema_checker = self.check_events_dict([
+            ('type', equals('realm_export')),
+            ('exports', check_list(check_dict_only([
+                ('id', check_int),
+                ('export_time', check_float),
+                ('acting_user_id', check_int),
+                ('export_url', equals(None)),
+                ('deleted_timestamp', equals(None)),
+                ('failed_timestamp', check_float),
+                ('pending', check_bool),
+            ]))),
+        ])
+
+        do_change_is_admin(self.user_profile, True)
+        self.login_user(self.user_profile)
+
+        with mock.patch('zerver.lib.export.do_export_realm',
+                        side_effect=Exception("test")):
+            with stdout_suppressed():
+                events = self.do_test(
+                    lambda: self.client_post('/json/export/realm'),
+                    state_change_expected=False, num_events=2)
+
+        error = pending_schema_checker('events[0]', events[0])
+        self.assert_on_error(error)
```

Comment:
Testing this in the original test function wasn't working as it was keeping the old state or something. I would see `export_url != None` and it would show the prior URL.  That's why this is split to its own function.